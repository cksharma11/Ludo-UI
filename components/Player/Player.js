import React from 'react';
import PropTypes from 'prop-types';
import Dice from '../Dice/Dice';
import PlayerStyles from './Player.style';
import noop from '../../config/utils/noop';
import Coin from '../Coin/Coin';
import labels from '../../config/labels/labels';

const Player = ({
  player,
  turn,
  alignment,
  onRollDice,
  diceValue,
  isWindowPlayer
}) => {
  const { coins } = player.coins;
  const { name } = player;
  const sortedCoins = coins.sort((a, b) => b.position - a.position);
  const clearedCoins = coins.filter(
    (coin) => coin.position === labels.CLEAR_COIN_POSITION
  );
  const coinsToWin = 4 - clearedCoins.length;

  const windowPlayerClass = isWindowPlayer ? 'window' : '';
  const activePlayerClass = turn ? 'current-turn' : '';
  const containerClass = `container ${alignment} ${windowPlayerClass} ${activePlayerClass} ${player.color}`;

  return (
    <div className={containerClass}>
      <div className="left-section">
        <div className="player-name">{name}</div>
        <div className="stat">{`${coinsToWin} coin(s) to win`}</div>
        {turn && (
          <Dice
            onRollDice={onRollDice}
            diceValue={diceValue}
            isWindowPlayer={isWindowPlayer}
          />
        )}
      </div>
      <div className="right-section" key="right-section">
        {sortedCoins.map((coin) => {
          const isClearedCoin = coin.position === labels.CLEAR_COIN_POSITION;
          return (
            <div key={`${coin.number}`} className="box">
              {isClearedCoin && <Coin color={player.color} />}
            </div>
          );
        })}
      </div>
      <style jsx>{PlayerStyles}</style>
    </div>
  );
};

Player.defaultProps = {
  turn: false,
  alignment: 'right',
  onRollDice: noop,
  diceValue: 1,
  isWindowPlayer: false
};

Player.propTypes = {
  turn: PropTypes.bool,
  alignment: PropTypes.string,
  onRollDice: PropTypes.func,
  diceValue: PropTypes.number,
  isWindowPlayer: PropTypes.bool,
  player: PropTypes.object.isRequired
};

export default Player;
