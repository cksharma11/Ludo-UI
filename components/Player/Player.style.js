import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const PlayerStyles = css`
  .container {
    display: flex;
    justify-content: space-between;
    margin: 10px 35px;
    width: 190px;
    border: 1px solid ${colors.gray.shade1};
    border-radius: 5px;
    padding: 20px;
  }

  .player-name {
    font-size: 20px;
  }

  .stat {
    font-size: 15px;
  }

  .red {
    background: rgb(231, 96, 96);
  }

  .green {
    background: rgb(112, 236, 112);
  }

  .yellow {
    background: rgb(230, 230, 91);
  }

  .blue {
    background: rgb(84, 84, 235);
  }

  .left {
    position: absolute;
    left: 0;
  }

  .right {
    position: absolute;
    right: 0;
  }

  .left-section,
  .right-section {
    display: flex;
    flex-direction: column;
  }

  .box {
    width: 25px;
    height: 25px;
    border: 1px solid ${colors.gray.shade1};
  }

  .coin-info {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
  }

  .coin-position {
    text-align: right;
    padding-left: 10px;
  }

  .active {
    background: white;
  }

  .clear {
    background: green;
  }
`;

export default PlayerStyles;
