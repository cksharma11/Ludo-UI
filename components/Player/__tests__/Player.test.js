import React from 'react';
import { shallow } from 'enzyme';
import Player from '../Player';

describe('Player', () => {
  let props;
  beforeEach(() => {
    props = {
      name: 'dummy',
      turn: false,
      coinData: [
        {
          color: 'red',
          isCleared: true
        },
        {
          color: 'red',
          isCleared: false
        },
        {
          color: 'red',
          isCleared: false
        },
        {
          color: 'red',
          isCleared: false
        }
      ]
    };
  });

  it('should render properly when turn is false', () => {
    const wrapper = shallow(<Player {...props} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should render properly when turn is true', () => {
    props.turn = true;
    const wrapper = shallow(<Player {...props} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
