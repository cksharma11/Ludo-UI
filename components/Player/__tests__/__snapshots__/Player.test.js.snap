// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should render properly when turn is false 1`] = `
"<div className=\\"jsx-3876423128 container right\\">
  <div className=\\"jsx-3876423128 left-section\\">
    <div className=\\"jsx-3876423128 player-name\\">
      dummy
    </div>
    <div className=\\"jsx-3876423128 stat\\">
      0 coins cleared
    </div>
  </div>
  <div className=\\"jsx-3876423128 right-section\\">
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
  </div>
  <JSXStyle id=\\"3876423128\\">
    .
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    m
    a
    r
    g
    i
    n
    :
    1
    0
    p
    x
     
    3
    5
    p
    x
    ;
    w
    i
    d
    t
    h
    :
    1
    9
    0
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    5
    p
    x
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    2
    0
    p
    x
    ;
    }
    .
    p
    l
    a
    y
    e
    r
    -
    n
    a
    m
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    2
    0
    p
    x
    ;
    }
    .
    s
    t
    a
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    p
    x
    ;
    }
    .
    r
    e
    d
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    1
    ,
    9
    6
    ,
    9
    6
    )
    ;
    }
    .
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    1
    1
    2
    ,
    2
    3
    6
    ,
    1
    1
    2
    )
    ;
    }
    .
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    0
    ,
    2
    3
    0
    ,
    9
    1
    )
    ;
    }
    .
    b
    l
    u
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    8
    4
    ,
    8
    4
    ,
    2
    3
    5
    )
    ;
    }
    .
    l
    e
    f
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    h
    e
    i
    g
    h
    t
    :
    1
    9
    0
    p
    x
    ;
    p
    o
    s
    i
    t
    i
    o
    n
    :
    a
    b
    s
    o
    l
    u
    t
    e
    ;
    l
    e
    f
    t
    :
    0
    ;
    }
    .
    r
    i
    g
    h
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    h
    e
    i
    g
    h
    t
    :
    1
    9
    0
    p
    x
    ;
    p
    o
    s
    i
    t
    i
    o
    n
    :
    a
    b
    s
    o
    l
    u
    t
    e
    ;
    r
    i
    g
    h
    t
    :
    0
    ;
    }
    .
    l
    e
    f
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    ,
    .
    r
    i
    g
    h
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    }
    .
    r
    i
    g
    h
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    }
    .
    b
    o
    x
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    w
    i
    d
    t
    h
    :
    2
    5
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    2
    5
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    }
    .
    c
    o
    i
    n
    -
    p
    o
    s
    i
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    r
    i
    g
    h
    t
    ;
    p
    a
    d
    d
    i
    n
    g
    -
    l
    e
    f
    t
    :
    1
    0
    p
    x
    ;
    }
    .
    a
    c
    t
    i
    v
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    w
    h
    i
    t
    e
    ;
    }
    .
    c
    l
    e
    a
    r
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    g
    r
    e
    e
    n
    ;
    }
  </JSXStyle>
</div>"
`;

exports[`Player should render properly when turn is true 1`] = `
"<div className=\\"jsx-3876423128 container right active\\">
  <div className=\\"jsx-3876423128 left-section\\">
    <div className=\\"jsx-3876423128 player-name\\">
      dummy
    </div>
    <div className=\\"jsx-3876423128 stat\\">
      0 coins cleared
    </div>
    <Dice onRollDice={[Function: noop]} diceValue={1} />
  </div>
  <div className=\\"jsx-3876423128 right-section\\">
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
    <div className=\\"jsx-3876423128 box\\" />
  </div>
  <JSXStyle id=\\"3876423128\\">
    .
    c
    o
    n
    t
    a
    i
    n
    e
    r
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    j
    u
    s
    t
    i
    f
    y
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    b
    e
    t
    w
    e
    e
    n
    ;
    m
    a
    r
    g
    i
    n
    :
    1
    0
    p
    x
     
    3
    5
    p
    x
    ;
    w
    i
    d
    t
    h
    :
    1
    9
    0
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    b
    o
    r
    d
    e
    r
    -
    r
    a
    d
    i
    u
    s
    :
    5
    p
    x
    ;
    p
    a
    d
    d
    i
    n
    g
    :
    2
    0
    p
    x
    ;
    }
    .
    p
    l
    a
    y
    e
    r
    -
    n
    a
    m
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    2
    0
    p
    x
    ;
    }
    .
    s
    t
    a
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    f
    o
    n
    t
    -
    s
    i
    z
    e
    :
    1
    5
    p
    x
    ;
    }
    .
    r
    e
    d
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    1
    ,
    9
    6
    ,
    9
    6
    )
    ;
    }
    .
    g
    r
    e
    e
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    1
    1
    2
    ,
    2
    3
    6
    ,
    1
    1
    2
    )
    ;
    }
    .
    y
    e
    l
    l
    o
    w
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    2
    3
    0
    ,
    2
    3
    0
    ,
    9
    1
    )
    ;
    }
    .
    b
    l
    u
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    r
    g
    b
    (
    8
    4
    ,
    8
    4
    ,
    2
    3
    5
    )
    ;
    }
    .
    l
    e
    f
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    h
    e
    i
    g
    h
    t
    :
    1
    9
    0
    p
    x
    ;
    p
    o
    s
    i
    t
    i
    o
    n
    :
    a
    b
    s
    o
    l
    u
    t
    e
    ;
    l
    e
    f
    t
    :
    0
    ;
    }
    .
    r
    i
    g
    h
    t
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    h
    e
    i
    g
    h
    t
    :
    1
    9
    0
    p
    x
    ;
    p
    o
    s
    i
    t
    i
    o
    n
    :
    a
    b
    s
    o
    l
    u
    t
    e
    ;
    r
    i
    g
    h
    t
    :
    0
    ;
    }
    .
    l
    e
    f
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    ,
    .
    r
    i
    g
    h
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    -
    m
    s
    -
    f
    l
    e
    x
    b
    o
    x
    ;
    d
    i
    s
    p
    l
    a
    y
    :
    f
    l
    e
    x
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    f
    l
    e
    x
    -
    d
    i
    r
    e
    c
    t
    i
    o
    n
    :
    c
    o
    l
    u
    m
    n
    ;
    }
    .
    r
    i
    g
    h
    t
    -
    s
    e
    c
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    -
    w
    e
    b
    k
    i
    t
    -
    b
    o
    x
    -
    p
    a
    c
    k
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    -
    w
    e
    b
    k
    i
    t
    -
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    -
    m
    s
    -
    f
    l
    e
    x
    -
    p
    a
    c
    k
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    j
    u
    s
    t
    i
    f
    y
    -
    c
    o
    n
    t
    e
    n
    t
    :
    s
    p
    a
    c
    e
    -
    a
    r
    o
    u
    n
    d
    ;
    }
    .
    b
    o
    x
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    w
    i
    d
    t
    h
    :
    2
    5
    p
    x
    ;
    h
    e
    i
    g
    h
    t
    :
    2
    5
    p
    x
    ;
    b
    o
    r
    d
    e
    r
    :
    1
    p
    x
     
    s
    o
    l
    i
    d
     
    #
    2
    2
    2
    8
    3
    1
    ;
    }
    .
    c
    o
    i
    n
    -
    p
    o
    s
    i
    t
    i
    o
    n
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    t
    e
    x
    t
    -
    a
    l
    i
    g
    n
    :
    r
    i
    g
    h
    t
    ;
    p
    a
    d
    d
    i
    n
    g
    -
    l
    e
    f
    t
    :
    1
    0
    p
    x
    ;
    }
    .
    a
    c
    t
    i
    v
    e
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    w
    h
    i
    t
    e
    ;
    }
    .
    c
    l
    e
    a
    r
    .
    j
    s
    x
    -
    3
    8
    7
    6
    4
    2
    3
    1
    2
    8
    {
    b
    a
    c
    k
    g
    r
    o
    u
    n
    d
    :
    g
    r
    e
    e
    n
    ;
    }
  </JSXStyle>
</div>"
`;
